hisat2_index="/mnt/Storage/home/zhaochengchen/Data/mm9/mm9"
refseq_gtf="/mnt/Storage/home/zhaochengchen/Data/mm9/mm9_refSeqsymbol_new.gtf"
refseq_bed=/mnt/Storage/home/zhaochengchen/Data/mm9/mm9.refseq.bed

data_path=/mnt/Storage/home/zhaochengchen/Work/1.MethylTransition/data/GSE65160_mmEmbryo_scRNAseq
log_path=/mnt/Storage/home/zhaochengchen/Work/1.MethylTransition/data/GSE65160_mmEmbryo_scRNAseq/log

zygote_e1_1a="SRR1772588"
zygote_e1_1b="SRR1772589"
zygote_e1_2a="SRR1772590"
zygote_e1_2b="SRR1772591"
c2_e1_1a="SRR1772592"
c2_e1_1b="SRR1772593"
c2_e1_2a="SRR1772594"
c2_e1_2b="SRR1772595"
c2_e2_1a="SRR1772596"
c2_e2_1b="SRR1772597"
c2_e2_2a="SRR1772598"
c2_e2_2b="SRR1772599"
c2_e3_1a="SRR1772600"
c2_e3_1b="SRR1772601"
c2_e3_2a="SRR1772602"
c2_e3_2b="SRR1772603"
c4_e1_1a="SRR1772604"
c4_e1_1b="SRR1772605"
c4_e1_2a="SRR1772606"
c4_e1_2b="SRR1772607"
c4_e1_3a="SRR1772608"
c4_e1_3b="SRR1772609"
c4_e1_4a="SRR1772610"
c4_e1_4b="SRR1772611"
c4_e2_1a="SRR1772612"
c4_e2_1b="SRR1772613"
c4_e2_2a="SRR1772614"
c4_e2_2b="SRR1772615"
c4_e2_3a="SRR1772616"
c4_e2_3b="SRR1772617"
c4_e2_4a="SRR1772618"
c4_e2_4b="SRR1772619"
c4_e3_1a="SRR1772620"
c4_e3_1b="SRR1772621"
c4_e3_2a="SRR1772622"
c4_e3_2b="SRR1772623"
c4_e3_3a="SRR1772624"
c4_e3_3b="SRR1772625"
c4_e3_4a="SRR1772626"
c4_e3_4b="SRR1772627"
c8_e1_1a="SRR1772628"
c8_e1_1b="SRR1772629"
c8_e1_2a="SRR1772630"
c8_e1_2b="SRR1772631"
c8_e1_3a="SRR1772632"
c8_e1_3b="SRR1772633"
c8_e1_4a="SRR1772634"
c8_e1_4b="SRR1772635"
c8_e1_5a="SRR1772636"
c8_e1_5b="SRR1772637"
c8_e1_6a="SRR1772638"
c8_e1_6b="SRR1772639"
c8_e1_7a="SRR1772640"
c8_e1_7b="SRR1772641"
c8_e1_8a="SRR1772642"
c8_e1_8b="SRR1772643"
c8_e2_1a="SRR1772644"
c8_e2_1b="SRR1772645"
c8_e2_2a="SRR1772646"
c8_e2_2b="SRR1772647"
c8_e2_3a="SRR1772648"
c8_e2_3b="SRR1772649"
c8_e2_4a="SRR1772650"
c8_e2_4b="SRR1772651"
c8_e2_5a="SRR1772652"
c8_e2_5b="SRR1772653"
c8_e2_6a="SRR1772654"
c8_e2_6b="SRR1772655"
c8_e2_7a="SRR1772656"
c8_e2_7b="SRR1772657"
c8_e2_8a="SRR1772658"
c8_e2_8b="SRR1772659"
c8_e3_1a="SRR1772660"
c8_e3_1b="SRR1772661"
c8_e3_2a="SRR1772662"
c8_e3_2b="SRR1772663"
c8_e3_3a="SRR1772664"
c8_e3_3b="SRR1772665"
c8_e3_4a="SRR1772666"
c8_e3_4b="SRR1772667"
c8_e3_5a="SRR1772668"
c8_e3_5b="SRR1772669"
c8_e3_6a="SRR1772670"
c8_e3_6b="SRR1772671"
c8_e3_7a="SRR1772672"
c8_e3_7b="SRR1772673"
c8_e3_8a="SRR1772674"
c8_e3_8b="SRR1772675"
ICM_1="SRR1799815"
ICM_2="SRR1799816"
ICM_3="SRR1799817"
ICM_4="SRR1799818"
ICM_5="SRR1799819"
TE_1="SRR1799820"
TE_2="SRR1799821"
TE_3="SRR1799822"
TE_4="SRR1799823"
TE_5="SRR1799824"

for sample in zygote_e1_1a zygote_e1_1b zygote_e1_2a zygote_e1_2b c2_e1_1a c2_e1_1b c2_e1_2a c2_e1_2b c2_e2_1a c2_e2_1b c2_e2_2a c2_e2_2b c2_e3_1a c2_e3_1b c2_e3_2a c2_e3_2b c4_e1_1a c4_e1_1b c4_e1_2a c4_e1_2b c4_e1_3a c4_e1_3b c4_e1_4a c4_e1_4b c4_e2_1a c4_e2_1b c4_e2_2a c4_e2_2b c4_e2_3a c4_e2_3b c4_e2_4a c4_e2_4b c4_e3_1a c4_e3_1b c4_e3_2a c4_e3_2b c4_e3_3a c4_e3_3b c4_e3_4a c4_e3_4b c8_e1_1a c8_e1_1b c8_e1_2a c8_e1_2b c8_e1_3a c8_e1_3b c8_e1_4a c8_e1_4b c8_e1_5a c8_e1_5b c8_e1_6a c8_e1_6b c8_e1_7a c8_e1_7b c8_e1_8a c8_e1_8b c8_e2_1a c8_e2_1b c8_e2_2a c8_e2_2b c8_e2_3a c8_e2_3b c8_e2_4a c8_e2_4b c8_e2_5a c8_e2_5b c8_e2_6a c8_e2_6b c8_e2_7a c8_e2_7b c8_e2_8a c8_e2_8b c8_e3_1a c8_e3_1b c8_e3_2a c8_e3_2b c8_e3_3a c8_e3_3b c8_e3_4a c8_e3_4b c8_e3_5a c8_e3_5b c8_e3_6a c8_e3_6b c8_e3_7a c8_e3_7b c8_e3_8a c8_e3_8b ICM_1 ICM_2 ICM_3 ICM_4 ICM_5 TE_1 TE_2 TE_3 TE_4 TE_5
do
	cd ~/Work/1.MethylTransition/data/GSE65160_mmEmbryo_scRNAseq/
	# mkdir ${sample}
	cd ${sample}
	# fastq-dump ${!sample}
	# wget ftp://ftp.sra.ebi.ac.uk/vol1/fastq/${!sample:0:6}/${!sample}/${!sample}.fastq.gz
	# cutadapt -q 30 -m 30 -g AAGCAGTGGTATCAACGCAGAGTACATGGG -G AAGCAGTGGTATCAACGCAGAGTACATGGG -a "A{100}" -a "T{100}" -A "A{100}" -A "T{100}" -o ${!sample}/${sample}_1.fastq.gz -p ${!sample}/${sample}_2.fastq.gz ${!sample}/${!sample}_1.clean.fq.gz ${!sample}/${!sample}_2.clean.fq.gz
	# cutadapt -q 30 -m 30 -g AAGCAGTGGTATCAACGCAGAGTACATGGG -a "A{100}" -a "T{100}" -o ${sample}_cut.fastq ${!sample}.fastq
	# nohup hisat2 -p 10 --dta -x ${hisat2_index} -U ${sample}_cut.fastq -S ${sample}.align.sam > ${sample}.align.log 2>&1

	# nohup hisat2 -p 15 --dta -x ${hisat2_index} -U ${!sample}.fastq -S ${sample}.align.sam > ${sample}.align.log 2>&1
	# nohup samtools sort -@ 15 -o ${sample}.bam ${sample}.align.sam > ${sample}.sam_sort.log 2>&1
	nohup /mnt/Storage/home/zhaochengchen/bin/stringtie-1.3.3b/stringtie -e -B -p 20 -G ${refseq_gtf} -o ${sample}.gtf ${sample}.bam > ${sample}.stringtie.log 2>&1

	# gzip ${!sample}.fastq
	# rm ${sample}_cut.fastq
	# rm ${sample}.align.sam
done


###################################################################################################
#####################################   1.read distribution   #####################################
###################################################################################################
# for sample in zygote_e1_1a zygote_e1_1b zygote_e1_2a zygote_e1_2b c2_e1_1a c2_e1_1b c2_e1_2a c2_e1_2b c2_e2_1a c2_e2_1b c2_e2_2a c2_e2_2b c2_e3_1a c2_e3_1b c2_e3_2a c2_e3_2b c4_e1_1a c4_e1_1b c4_e1_2a c4_e1_2b c4_e1_3a c4_e1_3b c4_e1_4a c4_e1_4b c4_e2_1a c4_e2_1b c4_e2_2a c4_e2_2b c4_e2_3a c4_e2_3b c4_e2_4a c4_e2_4b c4_e3_1a c4_e3_1b c4_e3_2a c4_e3_2b c4_e3_3a c4_e3_3b c4_e3_4a c4_e3_4b c8_e1_1a c8_e1_1b c8_e1_2a c8_e1_2b c8_e1_3a c8_e1_3b c8_e1_4a c8_e1_4b c8_e1_5a c8_e1_5b c8_e1_6a c8_e1_6b c8_e1_7a c8_e1_7b c8_e1_8a c8_e1_8b c8_e2_1a c8_e2_1b c8_e2_2a c8_e2_2b c8_e2_3a c8_e2_3b c8_e2_4a c8_e2_4b c8_e2_5a c8_e2_5b c8_e2_6a c8_e2_6b c8_e2_7a c8_e2_7b c8_e2_8a c8_e2_8b c8_e3_1a c8_e3_1b c8_e3_2a c8_e3_2b c8_e3_3a c8_e3_3b c8_e3_4a c8_e3_4b c8_e3_5a c8_e3_5b c8_e3_6a c8_e3_6b c8_e3_7a c8_e3_7b c8_e3_8a c8_e3_8b ICM_1 ICM_2 ICM_3 ICM_4 ICM_5 TE_1 TE_2 TE_3 TE_4 TE_5
# do
# 	cd ~/Work/1.MethylTransition/data/GSE65160_mmEmbryo_scRNAseq/
# 	read_distribution.py -i ${sample}/${sample}.bam -r ${refseq_bed} > ${log_path}/${sample}.readsDistribution
# 	echo ${sample}
# done

###################################################################################################
#############################   2.1 get mapping infomation               ##########################
#############################   2.2 get read distribution infomation     ##########################
#############################   2.3 gene_expression_matrix               ##########################
#############################   2.4 create gtf_file_path.txt             ##########################
###################################################################################################
python2 RNA_seq_preprocessing.py

python2 /mnt/Storage/home/zhaochengchen/bin/RemoveDuplicateGenes.py merge_fpkm.txt GSE65160_mmEmbryoFPKM.txt

###################################################################################################
#####################################   2.gene_count_matrix   #####################################
###################################################################################################
gtf_file_path=/mnt/Storage/home/zhaochengchen/Work/1.MethylTransition/data/GSE65160_mmEmbryo_scRNAseq/gtf_file_path.txt
path=/mnt/Storage/home/zhaochengchen/Work/1.MethylTransition/data/GSE65160_mmEmbryo_scRNAseq

python2 ~/bin/prepDE.py -i ${gtf_file_path} -g ${path}/gene_count.txt -t ${path}/transcript_count.txt > count_convertion.log 2>&1

sed -i "s/,/\t/g" ${path}/gene_count.txt
sed -i "s/,/\t/g" ${path}/transcript_count.txt

mv ${path}/gene_count.txt ${path}/GSE65160_mmEmbryoCounts.txt

###################################################################################################
#####################################     3.DEG detection     #####################################
###################################################################################################


# for sample in ICM_1 ICM_2 ICM_3 ICM_4 ICM_5 TE_1 TE_2 TE_3 TE_4 TE_5
# do
# 	samtools view ${sample}/${sample}.bam | gfold count -ann /mnt/Storage/home/zhaochengchen/Data/mm9/mm9_refSeq_symbol.gtf -tag stdin -o ${sample}/${sample}.read_cnt
# 	mv ${sample}.read_cnt ${sample}/${sample}.read_cnt
# done

# mkdir GFOLD
# cd GFOLD
# gfold diff -s1 ../ICM_1/ICM_1,../ICM_2/ICM_2,../ICM_3/ICM_3,../ICM_4/ICM_4,../ICM_5/ICM_5 -s2 ../TE_1/TE_1,../TE_2/TE_2,../TE_3/TE_3,../TE_4/TE_4,../TE_5/TE_5 -suf .read_cnt -o ICM_TE.diff

# mkdir cutoff_0.3
# mkdir cutoff_0.58
# for sample in ICM_TE
# do
# 	awk -v "OFS=\t" '{if ($3 >= 0.3) print $1}' ${sample}.diff  > cutoff_0.3/${sample}_up.deg
# 	awk -v "OFS=\t" '{if ($3 <= -0.3) print $1}' ${sample}.diff  > cutoff_0.3/${sample}_dn.deg
# 	awk -v "OFS=\t" '{if ($3 > -0.3 && $3 < 0.3) print $1}' ${sample}.diff  > cutoff_0.3/${sample}_no.deg
# 	grep -v "#" cutoff_0.3/${sample}_up.deg > cutoff_0.3/${sample}_up.geneSymbol
# 	grep -v "#" cutoff_0.3/${sample}_dn.deg > cutoff_0.3/${sample}_dn.geneSymbol
# 	grep -v "#" cutoff_0.3/${sample}_no.deg > cutoff_0.3/${sample}_no.geneSymbol
# 	wc -l cutoff_0.3/${sample}_up.geneSymbol
# 	wc -l cutoff_0.3/${sample}_dn.geneSymbol
# 	wc -l cutoff_0.3/${sample}_no.geneSymbol

# 	awk -v "OFS=\t" '{if ($3 >= 0.58) print $1}' ${sample}.diff  > cutoff_0.58/${sample}_up.deg
# 	awk -v "OFS=\t" '{if ($3 <= -0.58) print $1}' ${sample}.diff  > cutoff_0.58/${sample}_dn.deg
# 	awk -v "OFS=\t" '{if ($3 > -0.58 && $3 < 0.58) print $1}' ${sample}.diff  > cutoff_0.58/${sample}_no.deg
# 	grep -v "#" cutoff_0.58/${sample}_up.deg > cutoff_0.58/${sample}_up.geneSymbol
# 	grep -v "#" cutoff_0.58/${sample}_dn.deg > cutoff_0.58/${sample}_dn.geneSymbol
# 	grep -v "#" cutoff_0.58/${sample}_no.deg > cutoff_0.58/${sample}_no.geneSymbol
# 	wc -l cutoff_0.58/${sample}_up.geneSymbol
# 	wc -l cutoff_0.58/${sample}_dn.geneSymbol
# 	wc -l cutoff_0.58/${sample}_no.geneSymbol
# done

# rm */*.deg


